src/Simple.OData.Client.V4.Adapter/Metadata.cs

```diff
@@ -229,10 +229,24 @@ public override IEnumerable<string> GetDeclaredKeyPropertyNames(string collectio
                 return new string[] { };
 
             return entityType.DeclaredKey.Select(x => x.Name);
+        }
+
+        /// <summary>
+        /// Gets a collection of key name collections that represent the alternate keys of the given entity
+        /// </summary>
+        /// <see cref="https://github.com/OData/vocabularies/blob/master/OData.Community.Keys.V1.md"/>
+        /// <param name="collectionName">The collection name of the entity</param>
+        /// <returns>An enumeration of string enumerations representing the key names</returns>
+        public override IEnumerable<IEnumerable<string>> GetAlternateKeyPropertyNames(string collectionName)
+        {
+            var entityType = GetEntityType(collectionName);
+            return _model.GetAlternateKeysAnnotation(entityType)
+                .Select(x => x.Select(y => y.Key));
         }
 
         public override string GetFunctionFullName(string functionName)
         {
+            GetAlternateKeyPropertyNames("");
```

Comment:
Was this testing code, or is there a side effect here that is important?

Reply:
No idea how that got there, possible an accidental paste. Removed